src/interface/viewport.c

```diff
@@ -2632,13 +2634,321 @@ void store_interaction_info(paint_struct *ps)
 	}
 }
 
+static bool sub_679236_679662_679B0D_679FF1(uint32 ebx, rct_g1_element *image, uint8 *esi) {
+	if (!(image->flags & G1_FLAG_BMP)) {
+		assert(false);
+		return false;
+	}
+
+	if (ebx & 0x20000000) {
+		uint8 *ebx_palette = RCT2_GLOBAL(0x009ABDA4, uint8*);
+
+		uint8 al = *esi;
+		uint8 al2 = *(al + ebx_palette);
+
+		return (al2 != 0);
+	}
+
+	if (ebx & 0x40000000) {
+		return false;
+	}
+
+	return (*esi != 0);
+}
+
+/**
+ * rct2: 0x0067933B, 0x00679788, 0x00679C4A, 0x0067A117
+ */
+static bool sub_67933B_679788_679C4A_67A117(uint8 *esi, sint16 x_start_point, sint16 y_start_point, int round) {
+	const uint8 *ebx = esi + ((uint16 *) esi)[y_start_point];
+
+	uint8 last_data_line = 0;
+	while (!last_data_line) {
+		int no_pixels = *ebx++;
+		uint8 gap_size = *ebx++;
+
+		last_data_line = no_pixels & 0x80;
+
+		no_pixels &= 0x7F;
+
+		ebx += no_pixels;
+
+		if (round > 1) {
+			if (gap_size % 2) {
+				gap_size++;
+				no_pixels--;
+				if (no_pixels == 0) {
+					continue;
+				}
+			}
+		}
+
+		if (round == 4) {
+			if (gap_size % 4) {
+				gap_size += 2;
+				no_pixels -= 2;
+				if (no_pixels <= 0) {
+					continue;
+				}
+			}
+		}
+
+		int x_start = gap_size - x_start_point;
+		if (x_start <= 0) {
+			no_pixels += x_start;
+			if (no_pixels <= 0) {
+				continue;
+			}
+
+			x_start = 0;
+		} else {
+			// Do nothing?
+		}
+
+		x_start += no_pixels;
+		x_start--;
+		if (x_start > 0) {
+			no_pixels -= x_start;
+			if (no_pixels <= 0) {
+				continue;
+			}
+		}
+
+		if (ceil2(no_pixels, round) == 0) continue;
+
+		return true;
+	}
+
+	return false;
+}
+
 /**
  *
  *  rct2: 0x00679074
+ *
+ * @return 0x00141F569
  */
-void sub_679074(rct_drawpixelinfo *dpi, int imageId, int x, int y)
-{
-	RCT2_CALLPROC_X(0x00679074, 0, imageId, x, y, 0, (int)dpi, 0);
+static bool new_sub_679074(rct_drawpixelinfo *dpi, int imageId, sint16 x, sint16 y) {
+	rct_g1_element *image = &g1Elements[imageId & 0x7FFFF];
```

Comment:
we need to account for g2 elements as well for the future. Have a look at how its done in sprite draw

```
	rct_g1_element* g1_source;
	if (image_element < SPR_G2_BEGIN) {
		g1_source = &g1Elements[image_element];
	} else {
		g1_source = &g2.elements[image_element - SPR_G2_BEGIN];
	}
```